downstairs/src/lib.rs

```diff
@@ -1589,11 +1635,11 @@ impl Work {
      * Return a list of downstairs request IDs that are new or have
      * been waiting for other dependencies to finish.
      */
-    fn new_work(&self, upstairs_uuid: Uuid) -> Vec<u64> {
+    fn new_work(&self, upstairs_connection: UpstairsConnection) -> Vec<u64> {
         let mut result = Vec::with_capacity(self.active.len());
 
         for job in self.active.values() {
-            if job.upstairs_uuid != upstairs_uuid {
+            if job.upstairs_connection != upstairs_connection {
```

Comment:
This will check both `upstairs_id` and `session_id`?

Reply:
That's how my brain thought of this check, but actually, because UpstairsConnection automatically derives `PartialEq`, I believe this is also checking gen number.